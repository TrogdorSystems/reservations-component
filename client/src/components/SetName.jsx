import React from 'react';
import PropTypes from 'prop-types';

class SetName extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      nameVal: '',
    };
  }

  handleButtonClick() {
    if (this.state.nameVal.length > 0) {
      this.props.clickHandler(this.state.nameVal);
      this.setState({
        nameVal: '',
      });
    }
  }

  handleKeyPress(e) {
    if (e.key === 'Enter') {
      this.handleButtonClick();
    }
  }

  render() {
    const { name } = this.props;
    return (
      <div>
        {name
        ? `Welcome, ${name}`
        : (
          <div>
            <input
              type="text"
              placeholder="Enter name"
              id="nameInput"
              value={this.state.nameVal}
              onChange={(e) => { this.setState({ nameVal: e.target.value }); }}
              onKeyPress={e => this.handleKeyPress(e)}
            />
            <button
              id="nameButton"
              onClick={() => this.handleButtonClick()}
            >
              Enter Name
            </button>
          </div>)
      }
      </div>);
  }
}


SetName.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default SetName;
