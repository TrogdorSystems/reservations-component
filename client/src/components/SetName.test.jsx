import React from 'react';
import { shallow } from 'enzyme';

import SetName from './SetName';

describe('SetName Component', () => {
  test('should render correctly', () => {
    expect(shallow(<SetName />)).toMatchSnapshot();
  });
});
