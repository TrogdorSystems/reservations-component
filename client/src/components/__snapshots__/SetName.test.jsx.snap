// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetName Component should render correctly when a name is set 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SetName
    clickHandler={[Function]}
    name="Marge Simpson"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Welcome, Marge Simpson",
    },
    "ref": null,
    "rendered": "Welcome, Marge Simpson",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Welcome, Marge Simpson",
      },
      "ref": null,
      "rendered": "Welcome, Marge Simpson",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SetName Component should render correctly when no name is set 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SetName
    clickHandler={[Function]}
    name=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <input
          id="nameInput"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
        <button
          id="nameButton"
          onClick={[Function]}
        >
          Enter Name
        </button>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="nameInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter name"
            type="text"
            value=""
          />,
          <button
            id="nameButton"
            onClick={[Function]}
          >
            Enter Name
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "nameInput",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "Enter name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter Name",
            "id": "nameButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Enter Name",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <input
            id="nameInput"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter name"
            type="text"
            value=""
          />
          <button
            id="nameButton"
            onClick={[Function]}
          >
            Enter Name
          </button>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="nameInput"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Enter name"
              type="text"
              value=""
            />,
            <button
              id="nameButton"
              onClick={[Function]}
            >
              Enter Name
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "nameInput",
              "onChange": [Function],
              "onKeyPress": [Function],
              "placeholder": "Enter name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter Name",
              "id": "nameButton",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Enter Name",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
